[Cleanup Feature] - Form Validation

nikhithareddyv opened the pull request:
### Pivotal Tracker
- https://www.pivotaltracker.com/story/show/181085407

<!-- *** EDIT_NUMBER_INTO_LINK_ABOVE_REPLACING_### ***
    - Include a link to each user story on Pivotal for the user story, bugfix, etc.
    - If there is not an action item for the PR, then consider making one.
-->



### Description / PR Commit Message

<!-- *** COMMENT_DESCRIPTION_PR_COMMIT_MESSAGE ***
    - Add a description of the PR and why it is being made.
    - May contain root cause, perf impact, background info, system design, changes made, creator impact, etc.
    - Upon merging, copy the text you typed below here and use that as the PR Commit Message.
-->
Validates user sign up and form data


### Guidance for Review

<!-- *** COMMENT__GUIDANCE_FOR_REVIEW ***
    - Provide additional information to ease review of this PR; this section does not survive into git history.
    - May contain: what you dev tested, perf captures, screenshots, requested feedback, recommended order to review files in, auto test reliability results, changelog exemption reason, test review exemption reason, etc.
-->



### Author's Checklist
- [ ] Wrote tests?
- [ ] Linked to issue in Github?
- [ ] Tests passed?
- [ ] Test deployment to gatekeeper-tamu-test?

nikhithareddyv committed:
some user signup validation

nikhithareddyv committed:
some form validation

nikhithareddyv committed:
merge conflicts fixed

nikhithareddyv committed:
all search input types handled

nikhithareddyv committed:
Merge branch 'main' into nikhiv/validate_form_data

rebeccamcfadden commented on app/controllers/search_controller.rb:
```diff
@@ -29,11 +29,16 @@ def showsearch(title_name)
-                @titles = @idresult['title_results'][i]['name']
-                @search = Search.create(search_id: @id, title: @titles)
-                break
+            if @idresult['title_results'] == []
```
searches that cause errors:
- "house md": redirects to error page
- "avatar the last airbender": doesn't actually search
- "atla": returns empty results page

rebeccamcfadden replied:
please fix

rebeccamcfadden commented on app/models/user.rb:
```diff
@@ -23,12 +23,15 @@ def self.from_omniauth(auth)
 			:recoverable, :rememberable, :validatable,
 			:timeoutable, :omniauthable
 	validates :username, presence: true
+	validates :email, presence: true, length: { minimum: 10, maximum: 255 },format: { with: VALID_EMAIL_REGEX }
```
why is the minimum 10 characters for the email?

rebeccamcfadden replied:
also, validate presence of first and last name

rebeccamcfadden commented on app/views/subscriptions/_form.html.erb:
```diff
@@ -29,15 +29,15 @@
     <div class="field">
       <%= form.label :username %>
-      <%= form.text_field :username %>
+      <%= form.text_field :username, :required => 'required' %>
```
require subscription name as well

rebeccamcfadden replied:
also if you could remove the image field from this form that'd be great

rebeccamcfadden reviewed (changes requested):
left some comments. search in general is a bit buggy

also, please add validation for the temp_access email form

nikhithareddyv committed:
Search functionality improved

nikhithareddyv committed:
More codefactor fixes

nikhithareddyv committed:
failing cucumber test case fixed

nikhithareddyv committed:
hidden environment variables

nikhithareddyv committed:
some functionality changed

nikhithareddyv committed:
search tests still failing

nikhithareddyv committed:
merge
Merge branch 'nikhiv/validate_form_data' of https://github.com/gatekeeper-tamu/gatekeeper into nikhiv/validate_form_data

code-factor committed:
[CodeFactor] Apply fixes

Commit:
small test fix

Commit:
env fix for rspec

Commit:
other rspec fixes
